plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/api/TemplateConfigurationServices.java

```diff
@@ -72,20 +82,46 @@ public ConfigurationServices getConfigurationServices() {
      * @return generation eObject with definitions template information.
      */
     public Generation addProperties(Generation generation, TemplateInfo templateInfo) {
+        Multimap<String, EPackage> packagesByName = ArrayListMultimap.create();
+        for (String uri : templateInfo.getPackagesURIs()) {
+            if (EPackage.Registry.INSTANCE.containsKey(uri)) {
+                EPackage p = EPackage.Registry.INSTANCE.getEPackage(uri);
+                if (p != null) {
+                    packagesByName.put(p.getName(), p);
+                }
+            }
+        }
         for (String key : templateInfo.getVariables().keySet()) {
-            String type = templateInfo.getVariables().get(key);
+            String typeName = templateInfo.getVariables().get(key);
```

Comment:
Looking quickly over TemplateInfo it seems possible to have a variable which has a name but `null` as a type name.